modules/observable/src/main/scala/notebook/Codec.scala

```diff
@@ -1,49 +1,55 @@
 package notebook
 
-import scala.util.Try
+import java.util.Date
 
+import org.slf4j.LoggerFactory
 import play.api.libs.json._
-import play.api.libs.functional.syntax._
 
-import org.slf4j.LoggerFactory
+import scala.util.Try
 
-import notebook.util._
+trait Codec[A, B] {
+  def encode(x: A): B
 
-trait Codec[A,B] {
-  def encode(x:A):B
-  def decode(x:B):A
+  def decode(x: B): A
 }
 
 object JsonCodec {
-  val log = LoggerFactory.getLogger(getClass())
+  val log = LoggerFactory.getLogger(getClass)
 
-  implicit def formatToCodec[A](implicit f:Format[A]):Codec[JsValue, A] = new Codec[JsValue, A] {
-    def decode(a: A):JsValue = f.writes(a)
-    def encode(v: JsValue):A = f.reads(v) match {
+  implicit def formatToCodec[A](implicit f: Format[A]): Codec[JsValue, A] = new Codec[JsValue, A] {
+    def decode(a: A): JsValue = f.writes(a)
+
+    def encode(v: JsValue): A = f.reads(v) match {
       case s: JsSuccess[A] => s.get
       case e: JsError => throw new RuntimeException("Errors: " + JsError.toFlatJson(e).toString())
     }
   }
 
   //implicit val ints:Codec[JsValue, Int] = formatToCodec(Format.of[Int])
-  implicit val longs:Codec[JsValue, Long] = formatToCodec(Format.of[Long])
-  implicit val doubles:Codec[JsValue, Double] = formatToCodec(Format.of[Double])
-  implicit val floats:Codec[JsValue, Float] = formatToCodec(Format.of[Float])
-  implicit val strings:Codec[JsValue, String] = formatToCodec(Format.of[String])
-  implicit val chars:Codec[JsValue, Char] = formatToCodec(Format(Reads.of[String].map(_.head), Writes((c:Char) => JsString(c.toString))))
-  implicit val bools:Codec[JsValue, Boolean] = formatToCodec(Format.of[Boolean])
-
-  implicit def defaultDates(implicit d:java.text.DateFormat) = new Codec[JsValue, java.util.Date] {
-    def decode(t: java.util.Date):JsValue = JsNumber(t.getTime)
-    def encode(v: JsValue):java.util.Date = v match {
-      case JsString(s)  => Try(s.toLong).toOption.map(x => new java.util.Date(x)).getOrElse(d.parse(s))
+  implicit val longs: Codec[JsValue, Long] = formatToCodec(Format.of[Long])
+  implicit val doubles: Codec[JsValue, Double] = formatToCodec(Format.of[Double])
+  implicit val floats: Codec[JsValue, Float] = formatToCodec(Format.of[Float])
+  implicit val strings: Codec[JsValue, String] = formatToCodec(Format.of[String])
+  implicit val chars: Codec[JsValue, Char] = formatToCodec(Format(Reads.of[String].map(_.head), Writes((
```

Comment:
arf I know the line is long but the newline would be better at a different place